samza-api/src/main/java/org/apache/samza/table/descriptors/RemoteTableDescriptor.java

```diff
@@ -135,6 +137,7 @@ public RemoteTableDescriptor(String tableId) {
     Preconditions.checkNotNull(readFn, "null read function");
     Preconditions.checkNotNull(retryPolicy, "null retry policy");
     this.readRetryPolicy = retryPolicy;
+    this.readRetryPolicy.setTableId(String.format("%s.%s", tableId, READ_RETRY_POLICY));
```

Comment:
What happens when someone is reusing the same instance, would tableId be overwritten? Can we add a check to avoid this?

Reply:
Good point. In that case I think we can use a prefix "io.read.and.write.retry.policy" to indicate that this TableRetryPolicy is used for both read and write.
Another solution could be preventing users to use same instance. But that's not convenient for users.

Reply:
I don't think adding prefix will help. Well, I am afraid the later is the the only option.